app/src/main/java/com/grignon/shanghai/OnlineRoomFragment.java

```diff
@@ -0,0 +1,210 @@
+        roomTitleTextView.setText(roomTitle);
+
+        /* Set room code */
+        TextView roomCodeTextView = view.findViewById(R.id.roomCodeText);
+        roomCodeTextView.setText(room.getCode());
+
+        /* Set players number */
+        playerNumberTextView = view.findViewById(R.id.playerNumberText);
+        this.updatePlayerNumber(room);
+
+        /* Set recycler view data */
+        playersRecyclerView = view.findViewById(R.id.playersRecyclerView);
+        playersRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        this.updateRecyclerView(room);
+
+        /* Init socket which update players in the room */
+        this.socketRoomService = new SocketRoomService(this, this.onRoomUpdated, this.onLeaveRoom);
+
+        return view;
+    }
+
+    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        view.findViewById(R.id.exitButton).setOnClickListener(this::exit);
+        view.findViewById(R.id.playButton).setOnClickListener(this::play);
+    }
+
+    private void play(View view) {
+        if (room.getPlayers().size() < 2) {
+            Toast.makeText(getContext(), R.string.error_player_numbers, Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        this.isGameStarted = true;
+        this.socketRoomService.startGame(room);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (this.isGameStarted) {
+            return;
+        }
+
+        this.isAppStopped = true;
+        this.exit(getView());
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (!isAppStopped) {
+            return;
+        }
+
+        Bundle bundle = new Bundle();
+        bundle.putString("nickname", currentNickname);
+        NavSafeService.navigateSafe(
+                NavHostFragment.findNavController(this),
+                R.id.OnlineRoomFragment,
+                R.id.action_OnlineRoomFragment_to_OnlineRoomConnectionFragment
+        );
+    }
+
+    @Override
+    public void onDestroyView() {
+        this.exit(getView());
+        super.onDestroyView();
+    }
+
+    @Override
+    public void onDestroy() {
+        this.exit(getView());
+        super.onDestroy();
+    }
+
+    private void exit(View view) {
+        this.socketRoomService.disconnect(this.room, currentNickname);
+    }
+
+    public View.OnKeyListener onBackPressed = new View.OnKeyListener()
+    {
+        @Override
+        public boolean onKey( View v, int keyCode, KeyEvent event )
+        {
+            if( keyCode == KeyEvent.KEYCODE_BACK )
+                exit(v);
+
+            return false;
+        }
+    };
+
+    private final Emitter.Listener onRoomUpdated = new Emitter.Listener() {
+        @Override
+        public void call(Object... args) {
+            getActivity().runOnUiThread(new Runnable() {
```

Comment:
Utilse require pour éviter des NullPointerException 
```suggestion
            requireActivity().runOnUiThread(new Runnable() {
```